Bottom: 561cfb76d068a135005919275d3ed0d51e6c4e14
Top:    a61efc836c044069888cf21d104849f3947efc2d
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2008-11-02 17:08:34 -0500

Refresh of jds-lirc-mce2

---

diff --git a/drivers/input/ir/ir-mceusb2.c b/drivers/input/ir/ir-mceusb2.c
index 5e3b240..cc13e60 100644
--- a/drivers/input/ir/ir-mceusb2.c
+++ b/drivers/input/ir/ir-mceusb2.c
@@ -681,6 +681,8 @@ static void usb_remote_disconnect(struct usb_interface *intf)
 		return;
 
 	ir->usbdev = NULL;
+	input_ir_destroy(ir->input);
+	input_free_device(ir->input);
 
 	usb_kill_urb(ir->urb_in);
 	usb_free_urb(ir->urb_in);
